lightning/src/onion_message/messenger.rs

```diff
@@ -243,6 +243,54 @@ impl OnionMessageRecipient {
 	}
 }
 
+pub struct Responder<'a, OMH: OnionMessageHandler, T: OnionMessageContents> {
+	messenger: &'a OMH,
+	pub message: T,
+	pub reply_path: BlindedPath,
+	pub path_id: Option<[u8; 32]>
+}
+
+impl<'a, OMH: OnionMessageHandler, T: OnionMessageContents> Responder<'a, OMH, T> {
+    pub fn respond(&self, response: T) {
+        self.messenger.handle_onion_message_response(
+			response, self.reply_path.clone(), format_args!(
+				"when responding to {} onion message with path_id {:02x?}",
+				self.message.msg_type(),
+				self.path_id.clone()
+			)
+		);
+    }
+}
+
+pub struct MessageWithId<T: OnionMessageContents> {
+    pub message: T,
+    pub path_id: Option<[u8; 32]>,
+}
```

Comment:
Let's drop this in favor of using named fields in the enum variant.

```rust
WithoutReplyPath {
    message: T,
    path_id: Option<[u8; 32]>,
}
```

Reply:
Addressed in [8526afc](https://github.com/lightningdevkit/rust-lightning/pull/2907/commits/8526afc0989c69249a4fca1f10d03d15dc1a9c15)=
Thanks!